keps/sig-windows/0000-20181221-windows-group-managed-service-accounts-for-container-identity.md

```diff
@@ -99,105 +101,167 @@ A developer has a Windows web app that they would like to deploy in a container
+Next, GMSAAuthorizer will ensure the service account specified for the pod is authorized for a special `use` verb on the GMSACredSpec objects whose name is specified through the annotations or securityContext fields. GMSAAuthorizer will generate custom `AttributesRecord`s with `verb` set to `use`, `name` set to the GMSACredSpec object and `user` set to the service account of the pod. Finally, the `AttributesRecord`s will be passed to authorizers to check against RBAC configurations. A failure from the authorizes results in a 403: Forbidden.
 
-#### Changes in CRI's WindowsContainerSecurityContext struct
-A new field `CredentialSpec String` will be added to `WindowsContainerSecurityContext`. It will be populated by `generateWindowsContainerConfig` for Windows containers with the contents of key `gmsa-cred-spec` in the GMSA configmap reference from the pod (specified either through the annotation `pod.[alpha|beta].kubernetes.io/windows-gmsa-config-map` during alpha/beta stages or `securityContext.windows.credentialSpecConfig` in the pod spec at GA).
+During pod updates, changes to the credspec annotations will be blocked by GMSAAuthorizer and failed with 400: BadRequest.
+
+If the webhook is not installed and configured, no authorization checks will be performed on the contents of the credspec JSON. This will allow arbitrary credspec JSON to be specified for pods/containers and sent down to the container runtime.
+
+#### Changes in CRI's WindowsContainerSecurityContext struct and DetermineEffectiveSecurityContext
+
+A new field `CredentialSpec String` will be added to `WindowsContainerSecurityContext`. 
+
+`DetermineEffectiveSecurityContext` will be enhanced to populate `CredentialSpec` for Windows containers based on credspec specification for containers and pod. It will need to obtain the credspec contents from the `pod.alpha.kubernetes.io/windows-gmsa-credspec` and `container.windows-gmsa-credspec.alpha.kubernetes.io/container-name` annotations during the Alpha phase. The enhancements will be enabled under a feature flag during Alpha phase of the feature.
 
 #### Changes in Dockershim
-The `applyWindowsContainerSecurityContext` function will create a temporary file with a unique name on the host file system with the contents of the credspec. The file path will be used to populate `HostConfig.SecurityOpt` with a credspec file specification. The credspec file will be deleted after `StartContainer` in invoked as well as in various error handling paths between container creation and start.
+The `applyWindowsContainerSecurityContext` function will create a temporary file with a unique name on the host file system with the contents of `WindowsContainerSecurityContext.CredentialSpec`. The file path will be used to populate `HostConfig.SecurityOpt` with a credspec file specification. The credspec file will be deleted after `StartContainer` in invoked as well as in various error handling paths between container creation and start.
```

Comment:
What would be the path to the file?

Reply:
The file path will be under `C:\ProgramData\docker\CredentialSpecs\`. Will explain in the KEP.